src/compiler/generate-query-tests.xsl

```diff
@@ -261,7 +272,7 @@
          <xsl:text>      &#10;{&#10;</xsl:text>
          <xsl:choose>
             <xsl:when test="not($pending-p)">
-               <xsl:for-each select="$variables">
+               <xsl:for-each select="$variables[not(following-sibling::x:call)]">

```

Comment:
For testing purposes, I inserted `<xsl:message terminate="yes"/>` immediately after this `xsl:for-each` line (inside its block). All the tests passed. Is it because the tests do not cover this code path? Or is this code path just dormant?

Reply:
The code path is unused. The only caller, which is in `generate-common-tests.xsl`, passes in the `$variables` parameter as the `x:variable` elements that precede sibling `x:call` elements (recall that `x:context` does not occur for XQuery tests). My intention in lines 248-257 in `generate-query-tests.xsl` was to make sure that variables preceding the function call were defined early enough to potentially be used in the function call. Also, I didn't want to define each variable redundantly both before and after the function call. But by definition, `$variables` contains only the variables that precede the sibling function call, so there is no need for the additional processing of `$variables` after the function call. In c979b899029a0de64e9b17716342ff2ddac10296, I simplified the code and added a comment to clarify the contents of `$variables`.